# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessible.cpp
# Commit: 95497826aba2
# Full Hash: 95497826aba24fdf743d0926c26e074a3ec11f40
# Author: James Teh <jteh@mozilla.com>
# Date: 2024-09-04 09:55:13
# Regressor Bug: 1768396
# File Overlap Count: 2
# Description:
#   Bug 1768396: Cache IndexInParent on RemoteAccessible. r=nlapre
#   
#   Similar to LocalAccessible, IndexInParent is kept in an instance variable.
#   Upon insertion or removal, the index is updated on all impacted children.
#   
# ==============================================================================

diff -r 3138300cf7c8 -r 95497826aba2 accessible/ipc/RemoteAccessible.cpp
--- a/accessible/ipc/RemoteAccessible.cpp	Wed Sep 04 01:57:51 2024 +0000
+++ b/accessible/ipc/RemoteAccessible.cpp	Wed Sep 04 02:22:43 2024 +0000
@@ -83,6 +83,7 @@
   MOZ_ASSERT(aChildDoc);
   MOZ_ASSERT(mChildren.Length() == 0);
   mChildren.AppendElement(aChildDoc);
+  aChildDoc->mIndexInParent = 0;
 }
 
 void RemoteAccessible::ClearChildDoc(DocAccessibleParent* aChildDoc) {
@@ -93,6 +94,7 @@
   // ClearChildDoc() even though mChildren.Length() == 1.
   MOZ_ASSERT(mChildren.Length() <= 1);
   mChildren.RemoveElement(aChildDoc);
+  aChildDoc->mIndexInParent = -1;
 }
 
 uint32_t RemoteAccessible::EmbeddedChildCount() {